### ISSUE TITLE
1.5 Hours Course

### ISSUE THREAD

**Issue by anish-kum** [anish-kum/PPP-Spark-2025-2026#8]
90 min PPP

**Comment by anish-kum** [comment]
<img width="70" height="38" alt="Image" src="https://github.com/user-attachments/assets/58238a4e-dda1-449d-b1ab-e8f1011922bf" />

Start Time: 11:30 PM

**Comment by anish-kum** [comment]
<img width="336" height="662" alt="Image" src="https://github.com/user-attachments/assets/c84455e0-8fa3-48d6-b2ac-b9c4c9266789" />

<img width="324" height="222" alt="Image" src="https://github.com/user-attachments/assets/dca9ddcf-8936-4dfd-8fca-1f1dc181a64e" />


What I was learning:

- Error Function in Neural Networks: The error function quantifies how well the neural network's predictions match the actual labels, guiding the learning process.

- Backpropagation: This algorithm is used to compute the gradients of the loss function with respect to the weights, enabling efficient updates during training.

- Chain Rule: Backpropagation relies on the chain rule of calculus (taking Calculus BC right now) to propagate errors backward through the network layers.

- Learning Rate: Determines the size of the steps taken during optimization, influencing both convergence speed and stability.

- Overfitting: A common issue where the model learns noise in the training data instead of general patterns, leading to poor performance on unseen data.

- Regularization Techniques: Methods like L1 and L2 regularization are used to prevent overfitting by adding a penalty on the size of the weights.

- Validation Set: A separate dataset used to evaluate the model's performance during training, helping to tune hyperparameters and prevent overfitting.

One question the information didn't cover that I'd like an answer to: Apart from Calculus concepts, such as the Chain rule being used for concepts such as backpropagation, what other calculus concepts are used? The chain rule is a derivative concept, so do integrals ever come into play, and if so, when?

Note: Based on feedback from the PPP mentor last week, I want to shift my PPP to doing a hands-on project rather than a certificate if switching is allowed. Originally, I wasn't too sure if I should focus more on projects since I thought that learning AI theory is very important, but the mentor said that learning by doing would help accelerate my learning a lot so I'll probably take this week to figure out where to shift my focus to.

**Comment by anish-kum** [comment]
<img width="80" height="38" alt="Image" src="https://github.com/user-attachments/assets/c00b0646-df88-47c7-b697-741e37bc4c71" />

End time: 1:00 AM

**Comment by drewmca-pkwy** [comment]
@anish-kum Great question here, Anish. Yes, you may pivot. For this week, add a "Research" card. List out the websites you visit, the ideas you generate, and your process for thinking through what you might create. I'm looking forward to seeing what you find.
